imports/api/server/methods.tests.js

```diff
@@ -0,0 +1,160 @@
+import { Meteor } from 'meteor/meteor';
+import { assert, expect } from 'meteor/practicalmeteor:chai';
+import { resetDatabase } from 'meteor/xolvio:cleaner';
+import { Factory } from 'meteor/dburles:factory';
+import { student, company } from '/imports/api/seed/users.js';
+import { job } from '/imports/api/seed/jobs.js';
+import { sinon } from 'meteor/practicalmeteor:sinon';
+import '/imports/api/server/methods.js';
+
+describe('Application methods', () => {
+  beforeEach(() => {
+    resetDatabase();
+  });
+
+  describe('Inserting Company', () => {
+    it('successfully inserts a company into the users collection', () => {
+      let id = Meteor.call('saveCompany', company);
+      assert.equal(typeof(id), 'string', "should return user id");
+
+      let user = Meteor.users.find({_id: id}).fetch();
+      assert.equal(user.length, 1, "should list 1 user");
+    });
+
+    it('doesn\'t insert company into the users collection when document has errors', () => {
+      let result, error;
+      Meteor.call('saveCompany', {}, (err, res) => {
+        result = res;
+        error = err;
+      });
+
+      assert.equal(result, undefined, "doesn't return result");
+      assert.equal(error.message, "Need to set a username or email [400]", "return error message");
+    });
+  });
+
+  describe('Inserting Student', () => {
+    it('successfully inserts a student into the users collection', () => {
+      let id = Meteor.call('saveStudent', student);
+      assert.equal(typeof(id), 'string', "should return user id");
+
+      let user = Meteor.users.find({_id: id}).fetch();
+      assert.equal(user.length, 1, "should list 1 user");
+    });
+
+    it('doesn\'t insert student into the users collection when document has errors', () => {
+      let result, error;
+      Meteor.call('saveStudent', {}, (err, res) => {
+        result = res;
+        error = err;
+      });
+
+      assert.equal(result, undefined, "doesn't return result");
+      assert.equal(error.message, "Need to set a username or email [400]", "return error message");
+    });
+  });
+
+  describe('Executing job related methods', () => {
+    beforeEach(() => {
+      sinon.stub(Meteor, "user", () =>  company);
+      sinon.stub(Meteor, "userId", () =>  'id');
```

Comment:
This is what we call stubbing. When there is a method or function that we don't have access to it on tests we stub it.

Stubbing means faking the behavior of a method, fox example, in the tests we don't have access to the method `Meteor.user()`, because we would have to be logged to access it, so we are telling with `sinon.stub(Meteor, "user", () =>  company);` that when the user calls this method he should receive the value inside the variable `company`.

We create the stubs in the `beforeEach`, and destroy them in the `afterEach` method, to make sure we won't use to much space in the memory